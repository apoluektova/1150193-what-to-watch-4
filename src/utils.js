import {ALL_GENRES} from "./const.js";

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getGenresList = (movies) => {
  return [ALL_GENRES, ...new Set(movies.map((movie) => movie.genre))];
};


