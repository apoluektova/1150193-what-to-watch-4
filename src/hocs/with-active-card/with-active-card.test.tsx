import * as React from "react";
import * as renderer from "react-test-renderer";
import withActiveCard from "./with-active-card";
import {noop} from "../../utils";

const MockComponent = (props) => {
  return (
    <div {...props} />
  );
};

const MockComponentWrapped = withActiveCard(MockComponent);

it(`withActiveCard should render correctly`, () => {
  const tree = renderer
     .create(<MockComponentWrapped
       activeCard={{}}
       onActiveCardChange={noop}
     />)
     .toJSON();

  expect(tree).toMatchSnapshot();
});

