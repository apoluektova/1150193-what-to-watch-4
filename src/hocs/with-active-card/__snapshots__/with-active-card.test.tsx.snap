// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActiveCard should render correctly 1`] = `
<div
  activeCard={null}
  onActiveCardChange={[Function]}
/>
`;
